tests/v3_api/test_auth.py

```diff
@@ -98,13 +98,20 @@ def test_disable_and_enable_nestedgroups_set_access_control_required():
     validate_access_control_disable_and_enable_nestedgroups(access_mode)
 
 
-def test_disable_and_enable_nestedgroup_for_openldap_set_access_control_restricted():
+def test_disable_and_enable_nestedgroup_set_access_control_restricted():
     access_mode = "restricted"
     validate_access_control_disable_and_enable_nestedgroups(access_mode)
 
 
 def test_ad_service_account_login():
+    delete_project_users()
+    delete_cluster_users()
+    auth_setup_data = setup["auth_setup_data"]
+    admin_user = auth_setup_data["admin_user"]
+    admin_token = login(admin_user, PASSWORD)
     if AUTH_PROVIDER == "activeDirectory":
+        disable_ad(admin_user, admin_token)
+        enable_ad(admin_user, admin_token)
```

Comment:
does not enable_ad action make sure internally that service_account credential login works?

Reply:
discussed offline and the reason is we want to test this with a clean fresh Auth setup.

Reply:
I think it’s because the last test is validate_access_control_disable_and_enable_nestedgroups()
where the last user I logged in is users_under_nestedgroups
then in the next test, which is def test_ad_service_account_login():
I want to make the environment clear, and since i do not implement logout function, so I want to disable the old auth to make the user kind of log out, then enable auth